src/watch/mod.rs

```diff
@@ -1,15 +1,252 @@
 mod fs_watcher;
-mod main;
+mod migrate;
 
-pub use fs_watcher::FsWatcher;
-pub use main::run;
+pub use fs_watcher::{Event, FsWatcher};
+
+#[allow(unused_imports)]
+use crate::branding::{BRANDING_CLI_CMD, MANIFEST_FILE_DISPLAY_NAME};
+use crate::connect::Connector;
+use crate::options::Options;
+use crate::portable::project;
+use crate::print::{self, AsRelativeToCurrentDir, Highlight};
 
 #[derive(clap::Args, Debug, Clone)]
-pub struct WatchCommand {
-    #[command(flatten)]
-    pub conn: crate::options::ConnectionOptions,
+pub struct Command {
+    /// Runs "[BRANDING_CLI_CMD] migration apply --dev-mode" on changes to schema definitions.
+    /// On migration errors `force_database_error` is set, which rejects all queries
+    /// to the database. This configuration is cleared when the error is resolved or watch is stopped.
+    ///
+    /// This runs in addition to to scripts in [MANIFEST_FILE_DISPLAY_NAME].
+    #[arg(short = 'm', long)]
+    pub migrate: bool,
+
+    /// On failure, retry after a given number of seconds
+    #[arg(short = 'r', long)]
+    pub retry_sec: Option<u16>,
+
+    /// Don't suspend watch during script execution. Might cause loops.
+    #[arg(short = 'c', long)]
+    pub continuous: bool,
 
-    /// Print DDLs applied to the schema.
     #[arg(short = 'v', long)]
     pub verbose: bool,
 }
+
+#[tokio::main(flavor = "current_thread")]
+pub async fn run(options: &Options, cmd: &Command) -> anyhow::Result<()> {
+    let project = project::ensure_ctx_async(None).await?;
+    let mut ctx = WatchContext {
+        project,
+        connector: options.create_connector().await?,
+        is_force_database_error: false,
+    };
+
+    // determine what we will be watching
+    let mut matchers = assemble_matchers(cmd, &ctx)?;
+
+    if cmd.migrate {
+        print::msg!(
+            "Hint: --migrate will apply any schema changes to the database. When done use:"
```

Comment:
Maybe a bit more clear to point out that we're watching the file itself for changes? Otherwise "any schema changes" is a bit too ambiguous to understand what is happening, or what someone should do to trigger a "schema change".

How about:

```suggestion
            "Hint: --migrate will apply any changes in your schema module files to the database on save. When ready to commit your changes, use:"
```